### PR TITLE
Reduce overhead by fixing alignment in PerPriority

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead by fixing alignment in PerPriority
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/cluster_manager_impl.h [hunk 1]
```diff
@@ -371,13 +371,13 @@ class ClusterManagerImpl : public ClusterManager,
...
       PerPriority(uint32_t priority, const HostVector& hosts_added, const HostVector& hosts_removed)
-          : priority_(priority), hosts_added_(hosts_added), hosts_removed_(hosts_removed) {}
+          : hosts_added_(hosts_added), hosts_removed_(hosts_removed), priority_(priority) {}
 
-      const uint32_t priority_;
       const HostVector hosts_added_;
       const HostVector hosts_removed_;
       PrioritySet::UpdateHostsParams update_hosts_params_;
       LocalityWeightsConstSharedPtr locality_weights_;
+      const uint32_t priority_;
```
Comment (adisuissa) [inline source/common/upstream/cluster_manager_impl.h:380]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
Thanks!
LGTM modulo nit.

File: source/common/upstream/cluster_manager_impl.h (hunk 1 above)
Comment (DiazAlan) [reply source/common/upstream/cluster_manager_impl.h:381]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**